cirq/ops/projectors_test.py

```diff
@@ -0,0 +1,114 @@
+import re
+
+import numpy as np
+import pytest
+
+import cirq
+
+
+def test_projector_circuit_diagram():
+    q = cirq.NamedQubit('q')
+    projector = cirq.Projector([[1.0, 0.0]])
+    cirq.testing.assert_has_diagram(cirq.Circuit(projector(q)),
+                                    "q: ───Proj([[1.0, 0.0]])───",
+                                    precision=None)
+    cirq.testing.assert_has_diagram(cirq.Circuit(projector(q)),
+                                    "q: ───Proj([[1.0, 0.0]])───",
+                                    precision=2)
+
+
+def test_projector_qubit():
+    zero_projector = cirq.Projector([[1.0, 0.0]])
+    one_projector = cirq.Projector([[0.0, 1.0]])
+
+    np.testing.assert_allclose(cirq.channel(zero_projector),
+                               ([[1.0, 0.0], [0.0, 0.0]],))
+
+    np.testing.assert_allclose(cirq.channel(one_projector),
+                               ([[0.0, 0.0], [0.0, 1.0]],))
+
+
+def test_projector_from_np_array():
+    zero_projector = cirq.Projector(np.array([[1.0, 0.0]]))
+    np.testing.assert_allclose(cirq.channel(zero_projector),
+                               ([[1.0, 0.0], [0.0, 0.0]],))
+
+
+def test_projector_bad_rank():
+    with pytest.raises(ValueError,
+                       match="The input projection_basis must be a 2D array"):
+        cirq.Projector(np.array([1.0, 0.0]))
+
+
+def test_projector_plus():
+    plus_projector = cirq.Projector([[1.0, 1.0]])
```

Comment:
This should fail if we are checking for orthonormality...
This base vector does not have norm 1!! 
That would be `[1/np.sqrt(2), 1/np.sqrt(2)]`, right?

Reply:
Ignore this one - I see that orthonormality is not always enforced.